test/integration/BUILD

```diff
@@ -589,6 +589,7 @@ envoy_cc_test_library(
         ":http_protocol_integration_lib",
         ":socket_interface_swap_lib",
         "//source/common/http:header_map_lib",
+        "//source/extensions/clusters/logical_dns:logical_dns_cluster_lib",
```

Comment:
Similar question to the `//test/exe:main_common_test` target.
I guess it is ok because these are tests, but wonder if the extension is inherent to the tests functionality or is it testing non-core functionality (and then the test itself may be moved to the be part of the extension's tests)?